src/core/server/server.h

```diff
@@ -75,6 +76,37 @@
 #define GRPC_ARG_SERVER_MAX_PENDING_REQUESTS_HARD_LIMIT \
   "grpc.server.max_pending_requests_hard_limit"
 
+struct grpc_server_config_fetcher {
```

Comment:
Why do we have C-style names for new types?

Reply:
This isn't a new type, it's just moving the existing definition to a different place in the file.

Reply:
IIRC, all of the types that we're currently using `CppImplOf<>` for are single types, as opposed to cases like this one where it's an interface that will have subclasses.  I think we probably need to think through how `CppImplOf<>` will work for those cases.

A better option might be now that we can use C++ code in the C-core API, just change this API to be natively C++.  But we might want to do this in a separate PR, just because this one is already fairly large.

Reply:
This looks good.  I was thinking that `CppImplOf` would be problematic where there was a subclass type, but it doesn't look like that's actually a problem anywhere.